named_arrays/_functions/functions.py

```diff
@@ -45,6 +45,57 @@ def outputs(self) -> na.AbstractArray:
 
     __named_array_priority__: ClassVar[float] = 100 * na.AbstractVectorArray.__named_array_priority__
 
+    @property
+    def axes_center(self) -> tuple(str):
+        """
+        Return keys corresponding to all input axes representing bin centers
+        """
+        axes_center = tuple()
+        input_shape = self.inputs.shape
+        output_shape = self.outputs.shape
+        all_axes = set(input_shape | output_shape)
+
+        for axis in all_axes:
+            if axis in output_shape:
+                if axis in input_shape:
+                    if input_shape[axis] == output_shape[axis]:
+                        axes_center += (axis,)
+                    elif input_shape[axis] != output_shape[axis] + 1:
+                        raise ValueError(
+                            'Output axis dimension must either match Input axis dimension (representing'
+                            ' bin centers, or exceed by one (representing bin vertices).'
+                        )
+                else:
+                    axes_center += (axis,)
+
+            if axis in input_shape:
+                if axis not in output_shape:
+                    axes_center += (axis,)
+
+        return axes_center
+
+    @property
+    def axes_vertex(self) -> tuple(str):
+        """
+       Return keys corresponding to all input axes representing bin vertices
+       """
+        axes_vertex = tuple()
+        input_shape = self.inputs.shape
+        output_shape = self.outputs.shape
+        all_axes = set(input_shape | output_shape)
+
+        for axis in all_axes:
+            if axis in output_shape:
+                if axis in input_shape:
+                    if input_shape[axis] == output_shape[axis] + 1:
+                        axes_vertex += (axis,)
+                    elif input_shape[axis] != output_shape[axis]:
+                        raise ValueError(
+                            'Output axis dimension must either match Input axis dimension (representing'
+                            ' bin centers, or exceed by one (representing bin vertices).'
+                        )
```

Comment:
pretty sure we can just do this:

```suggestion
        axes_vertex = all_axes - self.axes_center
```

Reply:
Good point.  So should all_axes be a property or just recalculate it at the call to axes_vertex?

Reply:
Except apparently you can't just subtract sets and/or tuples.  I'll implement the spirit of the suggestion.